GeoKrety: search trackables with secret trackingCode not working (any more?)

While refactoring trackable logging I played around with GeoKrety for the first time.

I noticed that searching a GeoKrety using its secret tracking code does not work in c:geo. Is this a bug or is this intended behaviour?

I found code in c:geo indicating that at some point in the past such a search would work. 
It seems c:geo tries to get the geocode for a trackable by calling the following URL: `https://api.geokrety.org/nr2id/{secretTrackingCode}`. However, this URL returns HTTP-404
(Reference to c:geo code: `GeokretyConnector.getGeocodeFromTrackingCode`)

Also we have then to remove the test:
~~~
java.lang.AssertionError: Failed to get response from trackable website geokrety.org proxy https://api.geokrety.org
at cgeo.watchdog.WatchdogTest.checkWebsite(WatchdogTest.java:105)
at cgeo.watchdog.WatchdogTest.testTrackableWebsites(WatchdogTest.java:115)
~~~

--- a/main/src/main/java/cgeo/geocaching/connector/gc/GCConstants.java
+++ b/main/src/main/java/cgeo/geocaching/connector/gc/GCConstants.java
@@ -83,7 +83,8 @@ public final class GCConstants {
      */
     static final Pattern PATTERN_ATTRIBUTESINSIDE = Pattern.compile("<img src=\"([^\"]+)\" alt=\"([^\"]+?)\"");
     private static final String IMAGE_FORMATS = "jpg|jpeg|png|gif|bmp";
-    static final Pattern PATTERN_SPOILER_IMAGE = Pattern.compile("<a href=\"(https?://img(?:cdn)?\\.geocaching\\.com[^.]+\\.(?:" + IMAGE_FORMATS + "))\"[^>]+>" + "([^<]*)</a>" + ".*?(?:description\"[^>]*>([^<]+)</span>)?</li>", Pattern.DOTALL);
+    //Example HTML to match: <a href="https://img.geocaching.com/cache/large/1711f8a1-796a-405b-82ba-8685f2e9f024.jpg" class="owner-image" rel="owner_image_group" data-title="<strong>indy mit text netz Kopie</strong>">indy mit text netz Kopie</a></li>
+    static final Pattern PATTERN_SPOILER_IMAGE = Pattern.compile("<a href=\"(https?://img(?:cdn)?\\.geocaching\\.com[^.]+\\.(?:" + IMAGE_FORMATS + "))\"[^>]+>(?:[^>]+</strong>[^>]+>)?" + "([^<]*)</a>" + ".*?(?:description\"[^>]*>([^<]+)</span>)?</li>", Pattern.DOTALL);
     static final Pattern PATTERN_INVENTORY = Pattern.compile("ctl00_ContentBody_uxTravelBugList_uxInventoryLabel\">.*?WidgetBody(.*?)<div");
     static final Pattern PATTERN_INVENTORYINSIDE = Pattern.compile("[^<]*<li>[^<]*<a href=\"[a-z0-9\\-\\_\\.\\?\\/\\:\\@]*\\/(?:track|hide)\\/details\\.aspx\\?(guid|TB)=([0-9a-zA-Z\\-]+)[^\"]*\"[^>]*>[^<]*<img src=\"[^\"]+\"[^>]*>[^<]*<span>([^<]+)<\\/span>[^<]*<\\/a>[^<]*<\\/li>");
     static final Pattern PATTERN_WATCHLIST = Pattern.compile("data-cacheonwatchlist=\"True\"");

--- a/main/src/main/java/cgeo/geocaching/connector/trackable/GeokretyConnector.java
+++ b/main/src/main/java/cgeo/geocaching/connector/trackable/GeokretyConnector.java
@@ -28,6 +28,7 @@ import java.util.Collections;
 import java.util.List;
 import java.util.Locale;
 import java.util.TimeZone;
+import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 import org.apache.commons.collections4.CollectionUtils;
@@ -61,6 +62,8 @@ public class GeokretyConnector extends AbstractTrackableConnector {
     public static final String URL = "https://" + HOST;
     private static final String URLPROXY = "https://api." + HOST;
 
+    private static final Pattern ID_PATTERN = Pattern.compile("geokret id=\"([^\"]+)\"");
+
     @Override
     @NonNull
     public String getHost() {
@@ -91,8 +94,8 @@ public class GeokretyConnector extends AbstractTrackableConnector {
 
     @Override
     public boolean canHandleTrackable(@Nullable final String geocode, @Nullable final TrackableBrand brand) {
-        if (brand != TrackableBrand.GEOKRETY) {
-            return canHandleTrackable(geocode);
+        if (brand != null && brand != TrackableBrand.GEOKRETY && brand != TrackableBrand.UNKNOWN) {
+            return false;
         }
         return geocode != null && PATTERN_GK_CODE_EXTENDED.matcher(geocode).matches();
     }
@@ -296,12 +299,30 @@ public class GeokretyConnector extends AbstractTrackableConnector {
     @Nullable
     @WorkerThread
     private static String getGeocodeFromTrackingCode(final String trackingCode) {
-        final String response = Network.getResponseData(Network.getRequest(URLPROXY + "/nr2id/" + trackingCode));
+        //https://geokrety.org/api/v1/export2?tracking_code={trackingCode}
+        //Example response:
+        //<gkxml version="1.0" date="2024-08-07 19:02:22" date_Iso8601="2024-08-07T19:02:22+00:00">
+        //<geokrety>
+        //<geokret id="66184" type="1" collectible="false" owner_id="26422" ownername="kumy" holder_id="26422" holdername="" dist="36291" date="2017-12-13" places="12" image="GK10288_6503b68749e8a">
+        //<![CDATA[ Super Picsou Geant #200 ]]>
+        //</geokret>
+        //</geokrety>
+        //</gkxml>
+
+        final String response = Network.getResponseData(Network.getRequest(URL + "/api/v1/export2?tracking_code=" + trackingCode));
         // An empty response means "not found"
-        if (response == null || StringUtils.equals(response, "0")) {
+        if (response == null) {
             return null;
         }
-        return geocode(Integer.parseInt(response));
+        final Matcher m = ID_PATTERN.matcher(response);
+        if (m.find()) {
+            try {
+                return geocode(Integer.parseInt(m.group(1)));
+            } catch (NumberFormatException nfe) {
+                return null;
+            }
+        }
+        return null;
     }
 
     @Override

